Assignment7.txt

```diff
@@ -0,0 +1,22 @@
+//Anya Lauria 
+
+1. ++*p increments the variable that the pointer is pointing to first, and then evaluates it, similar to prefix 
+   incrementing in regular variables. *p++ evaluates the variable that the  pointer is pointing to first, then 
+   advances it. Not sure about *++p 
+2. The left to right order and right to left order depends on the kind of operation. For example, post fix operations 
+   always take precedence. The order can be manipulated using parantheses. 
+3. You can use them in function arguments, and not copy large chunks of memory, so the use of pointers is efficient. 
+   Pointers are also used to return arrays- there is no other way. Instead of using a character array, we can use 
+   pointers to work with strings instead, which is much easier. They also allow us to dynamic memory allocation, and
+   resize data structure whenever needed. 
+4.  4.1 char * 
+    4.2 invalid 
+    4.3 0==0 returns 1, so 1 
+    4.3 Int 
+    4.4 Int *- this is creating a pointer for a[0]
+    4.5 Int 
```

Comment:
```This actually returns the type int*. Recall that a[0] would be of type int and the ampersand just returns the memory address. So &a[0] would return the pointer to the address (since a pointer is basically an address, which is why all pointers are 8 bytes no matter what they're pointing to), which would be of type int*. ```